projects/packages/changelogger/src/SquashCommand.php

```diff
@@ -0,0 +1,275 @@
+use function Wikimedia\quietCall;
+
+/**
+ * "Squash" command for the changelogger tool CLI.
+ */
+class SquashCommand extends WriteCommand {
+
+	/**
+	 * The default command name.
+	 *
+	 * @var string|null
+	 */
+	protected static $defaultName = 'squash';
+
+	/**
+	 * Configures the command.
+	 */
+	protected function configure() {
+		$this->setDescription( 'Squash changelog entries in the changelog file' )
+			->addOption( 'regex', null, InputOption::VALUE_REQUIRED, 'PCRE regex to match the versions to squash, including delimiters' )
+			->addOption( 'count', null, InputOption::VALUE_REQUIRED, 'Number of versions to squash' )
+			->addOption( 'use-version', null, InputOption::VALUE_REQUIRED, 'Use this version instead of determining the version automatically' )
+			->addOption( 'release-date', null, InputOption::VALUE_REQUIRED, 'Release date, as a valid PHP date or "unreleased"' )
+			->addOption( 'prologue', null, InputOption::VALUE_REQUIRED, 'Prologue text for the new changelog entry' )
+			->addOption( 'epilogue', null, InputOption::VALUE_REQUIRED, 'Epilogue text for the new changelog entry' )
+			->addOption( 'link', null, InputOption::VALUE_REQUIRED, 'Link for the new changelog entry' )
+			->addOption( '--no-deduplicate', null, InputOption::VALUE_NONE, 'Do not deduplicate the changes' )
+			->addOption( 'yes', null, InputOption::VALUE_NONE, 'Default all questions to "yes" instead of "no". Particularly useful for non-interactive mode' )
+			->setHelp(
+				<<<EOF
+The <info>squash</info> command combines multiple existing changelog entries into one.
+
+The default behavior is to combine all entries that differ from the first only by prerelease
+and/or buildinfo. If <info>--regex</info> is used, instead all entries up to the first not matching
+the regex are merged. If <info>--count</info> is used, instead that many entries are merged.
+
+The default behavior is to combine the entries under the version number, date, and link of the
+first entry, with prologue and epilogue texts concatenated. This may be overridden using
+<info>--use-version</info>, <info>--release-date</info>, <info>--link</info>, <info>--prologue</info>,
+and <info>--epilogue</info>.
+EOF
+			);
+
+		try {
+			$this->getDefinition()->addOptions( Config::formatterPlugin()->getOptions() );
+		} catch ( \Exception $ex ) { // phpcs:ignore Generic.CodeAnalysis.EmptyStatement.DetectedCatch
+			// Will handle later.
+		}
+		try {
+			$this->getDefinition()->addOptions( Config::versioningPlugin()->getOptions() );
+		} catch ( \Exception $ex ) { // phpcs:ignore Generic.CodeAnalysis.EmptyStatement.DetectedCatch
+			// Will handle later.
+		}
+	}
+
+	/**
+	 * Deduplicate changes.
+	 *
+	 * @param InputInterface  $input InputInterface.
+	 * @param OutputInterface $output OutputInterface.
+	 * @param ChangeEntry[]   $changes Changes.
+	 * @return int
+	 */
+	protected function deduplicateChanges2( InputInterface $input, OutputInterface $output, array &$changes ) {
```

Comment:
I know there's another `deduplicateChanges` function in the WriteCommand.php - is the `2` appended here just to differentiate the functions when searching for them?

Reply:
They have different signatures, so (I assume) PHP would complain if I tried to give them the same name.

I suppose I could pass it the unneeded parameter and just ignore it, but meh.

Reply:
Nah it's all good, I was just curious since I had never seen the sequel to a function before 😂